apis/inv/v1alpha1/schema_helpers.go

```diff
@@ -66,9 +68,20 @@ func (r *SchemaSpec) GetNewSchemaBase(basePath string) SchemaSpecSchema {
 	basePath = r.GetBasePath(basePath)
 
 	// when no models are supplied we use the base dir
-	models := initSlice(r.Schema.Models, ".")
-	includes := initSlice(r.Schema.Includes, "")
-	excludes := initSlice(r.Schema.Excludes, "")
+	modelsSet := sets.New[string]()
+	includesSet := sets.New[string]()
+	excludesSet := sets.New[string]()
+	for _, repo := range r.Repositories {
+		mergeSetWithSlice(modelsSet, repo.Schema.Models)
+		mergeSetWithSlice(includesSet, repo.Schema.Includes)
+		mergeSetWithSlice(excludesSet, repo.Schema.Excludes)
+	}
+	models := sort.StringSlice(modelsSet.UnsortedList())
+	if len(models) == 0 {
+		models = []string{"."}
+	}
+	includes := sort.StringSlice(includesSet.UnsortedList())
+	excludes := sort.StringSlice(excludesSet.UnsortedList())
```

Comment:
https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#List
Returns a sorted list
```suggestion
	models := sets.List(modelsSet)
	if len(models) == 0 {
		models = []string{"."}
	}
	includes := sets.List(includesSet)
	excludes := sets.List(excludesSet)
```